interface/controller/pokemon_controller.go

```diff
@@ -0,0 +1,99 @@
+package controller
+
+import (
+	"net/http"
+	//"fmt"
+	//"errors"
+	"github.com/gorilla/mux"
+	"encoding/json"
+	"WizelineApi/domain/model"
+	"WizelineApi/service"
+	"strconv"
+)
+
+//Structura vacia del controlador. 
+type pokemonController struct{}
+
+type PokemonController interface{
+	GetAllPokemon()
+	GetPokemon()
+}
+
+
+
+func (pc *pokemonController) GetAllPokemons(res http.ResponseWriter, req *http.Request)  {
+	
+	pokemons := []model.Pokemon{}
+	err := service.ReadCSV("./test/bateria_csv/pokemon.csv", &pokemons)
+
+	if err != nil {
+		res.Header().Set("Content-Type", "application/json")
+		res.WriteHeader(http.StatusBadRequest)
+		res.Write([]byte(err.Error()))
+	}
+
+	//leer csv
+	// generar array del csv
+
+	res.Header().Set("Content-Type", "application/json")
+	res.WriteHeader(http.StatusOK)
+	json.NewEncoder(res).Encode(pokemons)
+}
+
+func (pc *pokemonController) GetPokemon(res http.ResponseWriter, req *http.Request) {
+	pokemons := []model.Pokemon{}
+	err := service.ReadCSV("./test/bateria_csv/pokemon.csv", &pokemons)
+
+	if err != nil {
```

Comment:
Could use a short declaration if statement combo here:
```suggestion
	if err := service.ReadCSV("./test/bateria_csv/pokemon.csv", &pokemons); err != nil {
```

Or having err and if lines cuddled one after another:
```suggestion
	err := service.ReadCSV("./test/bateria_csv/pokemon.csv", &pokemons)
	if err != nil {
```